helion/_compiler/device_ir.py

```diff
@@ -313,6 +313,28 @@ def build_rolled_reductions(self) -> None:
         for rdim in rdims:
             graph_to_info = {}
             allow_loop = False
+
+            # First, check if any graph contains matmul with rdim
+            # If so, we can't roll any graphs in this reduction dimension
+            can_roll_graphs = True
+            for graph_info in self.graphs:
+                roller = ReductionRoller(self, rdim, {})
+                if roller.has_matmul_with_rdim(graph_info.graph):
+                    can_roll_graphs = False
+                    break
+
+            if not can_roll_graphs:
+                # Can't roll any graphs for this rdim, but still need to add the spec
+                env.config_spec.reduction_loops.append(
+                    ReductionLoopSpec(
+                        block_id=rdim.block_id,
+                        size_hint=rdim.size_hint(),
+                    )
+                )
+                first = False
```

Comment:
Why are we adding the spec here?  The spec should only be added if things can be rolled. (After the latest refactor that removed boolean from the ReductionLoopSpec.)